### DISCUSSION TITLE
Is default value necessary now?

### DISCUSSION THREAD

**Discussion by cherrywins** [pydantic/pydantic#8427] [Pydantic V2]
After transition to Pydantic 2 I have a problem.
I want to sync some data.
View takes as argument data_in that should be type of SyncData.
For example
```
SyncData(BaseModel):
    field_1 str | None
    field_2 str | None
    field_3 str | None
    field_4 str | None

```
The data I send can look like this, considering that field_4 initially had another value than None:
```
{
    'field_1': 'value_1',
    'field_4': None,
}
```
In pydantic v1 I could send this data and there will be no error, validation would be passed. Now the pydantic says there are missing fields.
I changed my model:
```
SyncData(BaseModel):
    field_1 str | None = None
    field_2 str | None = None
    field_3 str | None = None
    field_4 str | None = None
```
Now I get in the update method not only initial 2 fields that I wanted to change, but other two as well. 
The filter data.model_dump(exclude_unset=True) doesn't work here because Pydantic sees default values as  a set values.
The filters exclude_defaults=True or exclude_none=True aren't suitable because I can send None or default value as a valid value to update the model. 

Is anyone has any ideas what I can do here?

**Comment by Viicos** [comment]
> In pydantic v1 I could send this data and there will be no error, validation would be passed. Now the pydantic says there are missing fields.

See https://docs.pydantic.dev/latest/migration/#required-optional-and-nullable-fields

> The filter data.model_dump(exclude_unset=True) doesn't work here because Pydantic sees default values as a set values.

I'm unable to reproduce with:

```python
from pydantic import BaseModel

class Model(BaseModel):
    field_1: str | None = None
    field_2: str | None = None

Model(field_1="test").model_dump(exclude_unset=True)
#> {'field_1': 'test'}  # No `field_2`
```
